core/src/banking_stage.rs

```diff
@@ -150,49 +151,72 @@ impl BankingStage {
         my_pubkey: &Pubkey,
         poh_recorder: &Arc<Mutex<PohRecorder>>,
         buffered_packets: &mut Vec<PacketsAndOffsets>,
+        id: u32,
     ) -> Result<UnprocessedPackets> {
         let mut unprocessed_packets = vec![];
         let mut rebuffered_packets = 0;
         let mut new_tx_count = 0;
+        let mut deserialized_count = 0;
         let buffered_len = buffered_packets.len();
         let mut buffered_packets_iter = buffered_packets.drain(..);
 
         let proc_start = Instant::now();
-        while let Some((msgs, unprocessed_indexes)) = buffered_packets_iter.next() {
+        while let Some((msgs, transactions, unprocessed_indexes)) = buffered_packets_iter.next() {
             let bank = poh_recorder.lock().unwrap().bank();
             if bank.is_none() {
                 rebuffered_packets += unprocessed_indexes.len();
-                Self::push_unprocessed(&mut unprocessed_packets, msgs, unprocessed_indexes);
+                Self::push_unprocessed(
+                    &mut unprocessed_packets,
+                    msgs,
+                    unprocessed_indexes,
+                    transactions,
+                );
                 continue;
             }
             let bank = bank.unwrap();
 
-            let (processed, verified_txs_len, new_unprocessed_indexes) =
+            let (processed, verified_txs_len, new_unprocessed_indexes, deserialized) =
                 Self::process_received_packets(
                     &bank,
                     &poh_recorder,
                     &msgs,
                     unprocessed_indexes.to_owned(),
+                    transactions.to_owned(),
```

Comment:
It would be nice if we could refactor to avoid this clone. Maybe a move inside the function and then have `process_received_packets` return those transactions that need to be retried?